raiden/tests/utils/smoketest.py

```diff
@@ -60,12 +62,45 @@
 if RST_GETH_BINARY is not None and 'RST_GETH_BINARY' not in os.environ:
     os.environ['RST_GETH_BINARY'] = RST_GETH_BINARY
 
-ports = iter(range(27854, 28000))
-# FIXME: get_free_port does not check for free ports (gh issue #759)
-get_free_port = lambda: str(next(ports))
 
-RST_RPC_PORT = get_free_port()
-os.environ['RST_RPC_PORT'] = RST_RPC_PORT
+def get_free_port(port_range):
```

Comment:
Please, merge this logic with: https://github.com/raiden-network/raiden/blob/master/raiden/tests/fixtures/variables.py#L201